core/store.go

```diff
@@ -288,12 +308,16 @@ func notifyWatchers(k string, operation string, obj *Obj) {
 // the deletion of the key.
 // This method is not thread-safe. It should be called within a lock.
 func deleteKey(k string, ptr unsafe.Pointer, obj *Obj) bool {
+	return deleteKeyWithOperationName(k, ptr, obj, "DEL")
+}
+
+func deleteKeyWithOperationName(k string, ptr unsafe.Pointer, obj *Obj, operation string) bool {
```

Comment:
As discussed in the other comment, we can stick to the original deleteKey implementation with `DEL` as the operation name being sent to the WatchChannel.